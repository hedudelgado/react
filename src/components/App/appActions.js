import { HeaderInterface, InvalidSessionErrorCodeMap, InvalidSessionHelper } from 'mega-code';
import {
  getCountries,
  getNewsletterRegions,
  getNewsletterPreferences,
  getProfile,
  updateProfile,
  deleteCustomer,
} from '../../utils/api';

// actions
export const USER_LOGGED_IN_SUCCESSFULLY = 'USER_LOGGED_IN_SUCCESSFULLY';
export const COUNTRIES_RECEIVED = 'COUNTRIES_RECEIVED';
export const PROFILE_RECEIVED = 'PROFILE_RECEIVED';
export const REGIONS_RECEIVED = 'REGIONS_RECEIVED';
export const NEWSLETTER_RECEIVED = 'NEWSLETTER_RECEIVED';
export const SET_EDIT_VIEW = 'SET_EDIT_VIEW';
export const SET_NOTIFICATION_MESSAGE = 'SET_NOTIFICATION_MESSAGE';
export const PROFILE_UPDATED = 'PROFILE_UPDATED';
export const NEWSLETTER_UPDATED = 'NEWSLETTER_UPDATED';
export const RESET_NEWSLETTER_INFO = 'RESET_NEWSLETTER_INFO';
export const RESET_USER_INFO = 'RESET_USER_INFO';
// views
export const DEFAULT_VIEW = undefined;
export const USER_DETAILS_VIEW = 'USER_DETAILS_VIEW';
export const PASSWORD_VIEW = 'PASSWORD_VIEW';
export const PAYMENT_CARD_VIEW = 'PAYMENT_CARD_VIEW';
export const ADD_GUEST_VIEW = 'ADD_GUEST_VIEW';
export const REGULAR_GUESTS_VIEW = 'REGULAR_GUESTS_VIEW';
export const NEWSLETTER_VIEW = 'NEWSLETTER_VIEW';
export const ROOM_REQUIREMENTS_VIEW = 'ROOM_REQUIREMENTS_VIEW';
export const EXTRAS_PREFERENCES_VIEW = 'EXTRAS_PREFERENCES_VIEW';
export const MEMORABLE_WORD_VIEW = 'MEMORABLE_WORD_VIEW';
// errors
export const PROFILE_ERROR = 'PROFILE_ERROR';
export const PASSWORD_ERROR = 'PASSWORD_ERROR';
export const NEWSLETTER_ERROR = 'NEWSLETTER_ERROR';
export const PAYMENT_CARD_ERROR = 'PAYMENT_CARD_ERROR';
export const USER_DETAILS_ERROR = 'USER_DETAILS_ERROR';
export const REGULAR_GUESTS_ERROR = 'REGULAR_GUESTS_ERROR';
export const MEMORABLE_WORD_ERROR = 'MEMORABLE_WORD_ERROR';
export const ROOM_REQUIREMENTS_ERROR = 'ROOM_REQUIREMENTS_ERROR';
export const EXTRAS_PREFERENCES_ERROR = 'EXTRAS_PREFERENCES_ERROR';

const checkForInvalidSession = (error, config, dispatch) => {
  if (InvalidSessionHelper
    .isSessionExpired(error, config)) {
    dispatch({
      type: RESET_USER_INFO,
    });
    dispatch({
      type: RESET_NEWSLETTER_INFO,
    });
    HeaderInterface.userLogOut(HeaderInterface.LOG_OUT_REASONS.SESSION_EXPIRED);
  }
};

export const getSettings = (email, sessionId, business) => (dispatch, getState) => {
  getProfile(email, sessionId, business).then((profile) => {
    dispatch({
      type: PROFILE_RECEIVED,
      profile,
    });
  }).catch((error) => {
    checkForInvalidSession(error, InvalidSessionErrorCodeMap.ACCOUNT.invalidSession, dispatch);
    dispatch({
      type: PROFILE_ERROR,
      error,
    });
  });
  getCountries()
    .then((countries) => {
      dispatch({
        type: COUNTRIES_RECEIVED,
        countries,
      });
    });
  if (getState().application.showEmailPreferences) {
    getNewsletterRegions()
      .then((regions) => {
        dispatch({
          type: REGIONS_RECEIVED,
          regions,
        });
      });

    getNewsletterPreferences(email)
      .then((newsletter) => {
        dispatch({
          type: NEWSLETTER_RECEIVED,
          newsletter,
        });
      })
      .catch(e => dispatch({
        type: NEWSLETTER_ERROR,
        error: e,
      }));
  }
};

export const updateUserProfile = payload => dispatch => updateProfile(payload).catch((error) => {
  checkForInvalidSession(error, InvalidSessionErrorCodeMap.ACCOUNT.invalidSession, dispatch);
  return Promise.reject(error);
});

export const deleteUserProfile = payload => dispatch => deleteCustomer(payload).catch((error) => {
  checkForInvalidSession(error, InvalidSessionErrorCodeMap.DELETE_PROFILE.invalidSession, dispatch);
  return Promise.reject(error);
});

export default {
  getSettings,
};
