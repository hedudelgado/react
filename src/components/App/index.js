export { default } from './App';

export { default as AppActions } from './appActions';

export { default as AppContainer } from './AppContainer';

export { default as modalReducer } from './modalReducer';
