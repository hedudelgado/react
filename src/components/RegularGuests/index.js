export { default } from './RegularGuests';

export { default as RegularGuestsContainer } from './RegularGuestsContainer';
