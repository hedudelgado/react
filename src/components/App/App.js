import React, { Component } from 'react';
import {
  array,
  object,
  func,
  bool,
  node,
} from 'prop-types';
import {
  PASSWORD_VIEW,
  PAYMENT_CARD_VIEW,
  USER_DETAILS_VIEW,
  ROOM_REQUIREMENTS_VIEW,
  EXTRAS_PREFERENCES_VIEW,
} from './appActions';
import { ComponentTogglerContainer } from '../ComponentToggler';
import UserProfile from '../UserProfile';
import Password from '../Password';
import UserProfileFormContainer from '../UserProfileForm';
import PasswordFormContainer from '../PasswordForm';
import { PaymentCardContainer } from '../PaymentCard';
import { RegularGuestsContainer } from '../RegularGuests';
import { NewsletterContainer } from '../Newsletter';
import Modal from '../../layout/Modal';
import PaymentCardFormContainer from '../PaymentCardForm';
import { DeleteCustomerContainer } from '../DeleteCustomer';
import RoomRequirements from '../RoomRequirements';
import RoomRequirementsFormContainer from '../RoomRequirementsForm';
import ExtrasPreferences from '../ExtrasPreferences';
import ExtrasPreferencesFormContainer from '../ExtrasPreferencesForm';
import BaseStyle from '../../layout/BaseStyle';
import MemorableWord from '../MemorableWord';
import MoveAccount from '../MoveAccount';
import {
  STAYER, SELF, managePermissionFlag, USER_LOGGED_IN_EVENT,
} from '../../config.json';
import ManagePermission from '../ManagePermission';

class App extends Component {
  constructor(props) {
    super(props);
    this.handleLoggedIn = this.handleLoggedIn.bind(this);
  }

  componentDidMount() {
    window.addEventListener('message', this.handleLoggedIn);
  }

  componentWillUnmount() {
    window.removeEventListener('message', this.handleLoggedIn);
  }

  async handleLoggedIn(event) {
    if (event.data.action === USER_LOGGED_IN_EVENT) {
      const { getSettings } = this.props;
      const { contactDetail: { email }, sessionId, business } = JSON.parse(event.data.profile);
      getSettings(email, sessionId, !!business);
    }
  }

  render() {
    const {
      countries,
      dictionary,
      profile,
      regions,
      newsletter,
      displayModal,
      handleModalClose,
      modalContent,
      userDataReceived,
      application: {
        showEmailPreferences,
      },
    } = this.props;

    const {
      business,
      business: {
        accessLevel = '',
        tetheredGuid = '',
      } = {},
    } = profile;

    const appReady = userDataReceived
      && (!showEmailPreferences || !!regions.length)
      && !!countries.length
      && (!showEmailPreferences || !!Object.keys(newsletter).length)
      && !!Object.keys(dictionary).length
      && !!Object.keys(profile).length;

    const showExtrasPreferences = !(accessLevel === SELF || accessLevel === STAYER);

    return (
      <div data-test="AccountSettings">
        {appReady
        && (
          <BaseStyle className="wb-typography app-container">
            <ComponentTogglerContainer
              viewName={USER_DETAILS_VIEW}
              view={UserProfile}
              edit={UserProfileFormContainer}
            />
            <ComponentTogglerContainer
              viewName={PASSWORD_VIEW}
              view={Password}
              edit={PasswordFormContainer}
            />
            <ComponentTogglerContainer
              viewName={PAYMENT_CARD_VIEW}
              view={PaymentCardContainer}
              edit={PaymentCardFormContainer}
            />
            {managePermissionFlag && (
              <ManagePermission dictionary={dictionary} />
            )}
            <RegularGuestsContainer />
            {showEmailPreferences && (
              <NewsletterContainer />
            )}
            <ComponentTogglerContainer
              viewName={ROOM_REQUIREMENTS_VIEW}
              view={RoomRequirements}
              edit={RoomRequirementsFormContainer}
            />
            {showExtrasPreferences && (
              <ComponentTogglerContainer
                viewName={EXTRAS_PREFERENCES_VIEW}
                view={ExtrasPreferences}
                edit={ExtrasPreferencesFormContainer}
              />
            )}
            {!!tetheredGuid && (
              <MemorableWord />
            )}
            {business && !tetheredGuid && (
              <MoveAccount />
            )}
            {!business && (
            <DeleteCustomerContainer />
            )}
            {displayModal && <Modal data-test="Modal" display content={modalContent} handleClose={handleModalClose} />}
          </BaseStyle>
        )
      }
      </div>
    );
  }
}

App.propTypes = {
  countries: array.isRequired,
  dictionary: object.isRequired,
  profile: object.isRequired,
  getSettings: func.isRequired,
  displayModal: bool.isRequired,
  handleModalClose: func.isRequired,
  modalContent: node,
  newsletter: object.isRequired,
  regions: array.isRequired,
  userDataReceived: bool.isRequired,
  application: object.isRequired,
};

export default App;
