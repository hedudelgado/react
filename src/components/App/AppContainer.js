import { connect } from 'react-redux';
import { getSettings as getSettingsAction } from './appActions';
import { closeModal } from './modalActions';

import App from './App';

const mapStateToProps = ({
  modal: {
    display: displayModal,
    content: modalContent,
  },
  dictionary,
  config,
  countries,
  regions,
  newsletter,
  application,
  userProfile: {
    userDataReceived,
    profile,
  },
}) => ({
  profile,
  displayModal,
  modalContent,
  dictionary,
  config,
  countries,
  regions,
  newsletter,
  application,
  userDataReceived,
});

const mapDispatchToProps = dispatch => ({
  getSettings:
    (email, sessionId, business) => dispatch(getSettingsAction(email, sessionId, business)),
  handleModalClose: () => dispatch(closeModal()),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
