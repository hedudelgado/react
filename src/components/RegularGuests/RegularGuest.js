import React, { Component } from 'react';
import {
  string,
  func,
  number,
  object,
} from 'prop-types';
import Button from '../../layout/Button';
import Typography from '../../layout/Typography';
import BaseStyle from '../../layout/BaseStyle';
import DeleteModal from '../../layout/DeleteModal';
import { REGULAR_GUESTS_ERROR, REGULAR_GUESTS_VIEW } from '../App/appActions';

class RegularGuest extends Component {
  constructor(props) {
    super(props);

    this.deleteGuest = this.deleteGuest.bind(this);
  }

  async deleteGuest() {
    const {
      profile,
      closeModal,
      setNotificationMessage,
      updateProfileStore,
      profile: { additionalGuests },
      index,
      updateProfile,
    } = this.props;

    const payload = {
      ...profile,
      additionalGuests: [
        ...additionalGuests,
      ],
    };

    payload.additionalGuests.splice(index, 1);

    await updateProfile(payload)
      .then(() => {
        updateProfileStore(payload);
        setNotificationMessage(REGULAR_GUESTS_VIEW);
      }).catch(() => {
        setNotificationMessage(REGULAR_GUESTS_ERROR);
      });

    closeModal();
  }

  showEditMode(i) {
    const { setEditView, hideSuccessMessage } = this.props;
    setEditView(`${REGULAR_GUESTS_VIEW}_${i}`);
    hideSuccessMessage();
  }

  render() {
    const {
      dictionary,
      title,
      firstName,
      lastName,
      email,
      index,
      openModal,
      closeModal,
    } = this.props;

    return (
      <BaseStyle component="address" mb="3">
        <Typography component="span">{`${title} ${firstName} ${lastName}`}</Typography>
        <Typography component="span">{email}</Typography>
        <Button
          id={`delete-guest-${index}`}
          className="wb-text-btn--underline block mt2 mb2"
          color="red"
          variant="text"
          onClick={() => openModal(
            <DeleteModal
              title={dictionary['guests.delete.title']}
              subtext={dictionary['guests.delete.subtext']}
              deleteFunc={this.deleteGuest}
              buttonLabel={dictionary['guests.delete.button.label']}
              handleClose={closeModal}
              cancelLabel={dictionary['guests.cancel.link']}
            />,
          )}
        >
          {`${dictionary['guests.delete.link']} ${index + 1}`}
        </Button>
        <Button data-test={`edit-guest-${index}`} color="quaternary" onClick={() => this.showEditMode(index)}>
          {`${dictionary['guests.edit.button']} ${index + 1}`}
        </Button>
      </BaseStyle>
    );
  }
}

RegularGuest.propTypes = {
  dictionary: object.isRequired,
  profile: object.isRequired,
  updateProfileStore: func.isRequired,
  title: string.isRequired,
  firstName: string.isRequired,
  lastName: string.isRequired,
  email: string.isRequired,
  setNotificationMessage: func.isRequired,
  index: number.isRequired,
  openModal: func.isRequired,
  closeModal: func.isRequired,
  setEditView: func.isRequired,
  hideSuccessMessage: func.isRequired,
  updateProfile: func.isRequired,
};

export default RegularGuest;
