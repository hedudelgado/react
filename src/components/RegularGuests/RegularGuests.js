import React from 'react';
import { func, object } from 'prop-types';
import Typography from '../../layout/Typography';
import { ComponentTogglerContainer } from '../ComponentToggler';
import {
  REGULAR_GUESTS_VIEW,
  ADD_GUEST_VIEW,
  REGULAR_GUESTS_ERROR,
} from '../App/appActions';
import RegularGuest from './RegularGuest';
import AddRegularGuest from '../AddRegularGuest';
import RegularGuestsForm from '../RegularGuestsForm';
import Notification from '../../layout/Notification';

const MAX_REGULAR_GUESTS = 6;

const RegularGuests = ({
  profile: {
    additionalGuests = [],
  },
  dictionary,
  app: { notificationMessage },
  openModal,
  closeModal,
  setEditView,
  hideSuccessMessage,
}) => {
  const userHasGuests = additionalGuests && additionalGuests.length;

  return (
    <section className="component-container">
      {notificationMessage === REGULAR_GUESTS_ERROR && (
        <Notification data-test="error-notification" priority={2} variant="error" ariaLive="polite" show>
          {dictionary['generic.error']}
        </Notification>
      )}
      {notificationMessage === REGULAR_GUESTS_VIEW && (
        <Notification data-test="success-message" priority={2} variant="success" ariaLive="polite" show>
          {dictionary['guests.notification.text']}
        </Notification>
      )}
      <Typography component="h4" className="pi-color-greyscale-darkest">
        {dictionary['guests.title']}
      </Typography>
      {!userHasGuests && (
      <Typography mb="3" data-test="no-guests-message">
        {dictionary['guests.noGuests']}
      </Typography>
      )}
      {
          !!userHasGuests
          && additionalGuests.map((guest, i) => (
            <ComponentTogglerContainer
              key={guest.email}
              viewName={`${REGULAR_GUESTS_VIEW}_${i}`}
              view={RegularGuest}
              edit={RegularGuestsForm}
              {...guest}
              index={i}
              setEditView={setEditView}
              hideSuccessMessage={hideSuccessMessage}
              openModal={openModal}
              closeModal={closeModal}
            />
          ))
      }
      { additionalGuests.length < MAX_REGULAR_GUESTS && (
        <ComponentTogglerContainer
          viewName={ADD_GUEST_VIEW}
          view={AddRegularGuest}
          edit={RegularGuestsForm}
        />
      )}
    </section>
  );
};

RegularGuests.propTypes = {
  app: object.isRequired,
  profile: object.isRequired,
  dictionary: object.isRequired,
  hideSuccessMessage: func.isRequired,
  openModal: func.isRequired,
  closeModal: func.isRequired,
  setEditView: func.isRequired,
};

export default RegularGuests;
