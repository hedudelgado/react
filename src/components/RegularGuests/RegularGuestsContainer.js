import { connect } from 'react-redux';
import RegularGuests from './RegularGuests';
import {
  DEFAULT_VIEW,
  SET_EDIT_VIEW,
  SET_NOTIFICATION_MESSAGE,
  REGULAR_GUESTS_VIEW,
} from '../App/appActions';
import { openModal, closeModal } from '../App/modalActions';


const mapStateToProps = ({
  dictionary,
  userProfile: { profile },
  app,
}) => ({
  dictionary,
  profile,
  app,
});

const mapDispatchToProps = dispatch => ({
  setEditView: viewName => dispatch({
    type: SET_EDIT_VIEW,
    editViewName: viewName,
  }),
  toggleEditView: (i = 0) => dispatch({
    type: SET_EDIT_VIEW,
    editViewName: `${REGULAR_GUESTS_VIEW}_${i}`,
  }),
  hideSuccessMessage: () => dispatch({
    type: SET_NOTIFICATION_MESSAGE,
    notificationMessage: DEFAULT_VIEW,
  }),
  openModal: content => dispatch(openModal(content)),
  closeModal: () => dispatch(closeModal()),
});

export default connect(mapStateToProps, mapDispatchToProps)(RegularGuests);
