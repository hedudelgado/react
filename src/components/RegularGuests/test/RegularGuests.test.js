import React from 'react';
import { shallow } from 'enzyme';
import { DEFAULT_VIEW, REGULAR_GUESTS_ERROR, REGULAR_GUESTS_VIEW } from '../../App/appActions';
import RegularGuests from '../RegularGuests';

describe('RegularGuests Component', () => {
  let wrapper;

  const props = {
    profile: {
      additionalGuests: [],
    },
    dictionary: {
      'guests.title': 'Regular Guests',
      'guests.noGuests': 'There are no guests',
      'guests.button.add': 'Add guest',
      'guests.button.cancel': 'Cancel changes',
      'guests.notification.text': 'Your details have been updated successfully!',
    },
    app: {},
    setEditView: jest.fn(),
    hideSuccessMessage: jest.fn(),
    openModal: jest.fn(),
    closeModal: jest.fn(),
    index: 1,
    showEditMode: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(<RegularGuests {...props} />);
  });

  describe('Renders RegularGuests Component Elements', () => {
    it('should render the component', () => {
      expect(wrapper).not.toBeEmptyRender();
    });

    it('should show success message when guests have updated successfully', () => {
      const copyProps = {
        ...props,
        app: { editViewName: DEFAULT_VIEW, notificationMessage: REGULAR_GUESTS_VIEW },
      };
      wrapper = shallow(<RegularGuests {...copyProps} />);
      expect(wrapper).toContainExactlyOneMatchingElement('[data-test="success-message"]');
    });

    it('should show error message when api fails', () => {
      const copyProps = {
        ...props,
        app: { editViewName: DEFAULT_VIEW, notificationMessage: REGULAR_GUESTS_ERROR },
      };
      wrapper = shallow(<RegularGuests {...copyProps} />);
      expect(wrapper).toContainExactlyOneMatchingElement('[data-test="error-notification"]');
    });

    it('should not show the Add Guest Button if the account has the max number of regular guests', () => {
      const fullGuests = {
        ...props,
        profile: {
          additionalGuests: [
            {
              title: 'Mr',
              firstName: 'Mac',
              lastName: 'Donalds',
              email: 'mac@test.com',
              telephone: '1111111111',
              mobile: '',
              nationality: 'GB',
              carRegistration: '',
            },
            {
              title: 'Ms',
              firstName: 'Burger',
              lastName: 'King',
              email: 'bk@test.com',
              telephone: '1111111111',
              mobile: '',
              nationality: 'GB',
              carRegistration: '',
            },
            {
              title: 'Mrs',
              firstName: 'Pizza',
              lastName: 'Hut',
              email: 'ph@test.com',
              telephone: '1111111111',
              mobile: '',
              nationality: 'GB',
              carRegistration: '',
            },
            {
              title: 'Miss',
              firstName: 'Kentucky',
              lastName: 'Fried Chicken',
              email: 'kfc@test.com',
              telephone: '1111111111',
              mobile: '',
              nationality: 'GB',
              carRegistration: '',
            },
            {
              title: 'Mrs',
              firstName: 'Jolibee',
              lastName: 'Chicken',
              email: 'jc@test.com',
              telephone: '1111111111',
              mobile: '',
              nationality: 'GB',
              carRegistration: '',
            },
            {
              title: 'Miss',
              firstName: 'Choo',
              lastName: 'Choo',
              email: 'cc@test.com',
              telephone: '1111111111',
              mobile: '',
              nationality: 'GB',
              carRegistration: '',
            },
          ],
        },
      };

      wrapper = shallow(<RegularGuests {...fullGuests} />);

      expect(wrapper.find('#add-guest')).toHaveLength(0);
    });
  });

  describe('Renders RegularGuests Component Data', () => {
    it('should contain a message if there are no regular guests', () => {
      expect(wrapper).toContainExactlyOneMatchingElement('[data-test="no-guests-message"]');
    });

    it('should have two REGULAR_GUESTS_VIEWs, one for each guest', () => {
      props.profile.additionalGuests = [
        {
          title: 'Mr',
          firstName: 'Mac',
          lastName: 'Donalds',
          email: 'mac@test.com',
          telephone: '1111111111',
          mobile: '',
          nationality: 'GB',
          carRegistration: '',
        },
        {
          title: 'Ms',
          firstName: 'Burger',
          lastName: 'King',
          email: 'bk@test.com',
          telephone: '1111111111',
          mobile: '',
          nationality: 'GB',
          carRegistration: '',
        },
      ];
      wrapper = shallow(<RegularGuests {...props} />);
      expect(wrapper).toContainMatchingElements(2, '[viewName^="REGULAR_GUESTS_VIEW_"]');
    });

    it('should default to empty array', () => {
      const regularGuestsProps = {
        ...props,
      };

      delete regularGuestsProps.profile.additionalGuests;

      const newRegularGuestsWrapper = shallow(<RegularGuests {...regularGuestsProps} />);
      expect(newRegularGuestsWrapper).toMatchSnapshot();
    });
  });
});
