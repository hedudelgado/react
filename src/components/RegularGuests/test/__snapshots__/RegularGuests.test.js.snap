// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegularGuests Component Renders RegularGuests Component Data should default to empty array 1`] = `
<section
  className="component-container"
>
  <Typography
    className="pi-color-greyscale-darkest"
    component="h4"
  >
    Regular Guests
  </Typography>
  <Typography
    component="p"
    data-test="no-guests-message"
    mb="3"
  >
    There are no guests
  </Typography>
  <Connect(ComponentToggler)
    edit={[Function]}
    view={[Function]}
    viewName="ADD_GUEST_VIEW"
  />
</section>
`;
